### PR TITLE
Dockerize goreleaser

### PR DESCRIPTION
**Pull request by richardartoul**

### CODE REVIEW

File: docker/goreleaser/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM goreleaser/goreleaser
```
Comment (prateek) [inline docker/goreleaser/Dockerfile]: specify an exact version please.

File: docker/goreleaser/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+COPY . /m3
```
Comment (prateek) [inline docker/goreleaser/Dockerfile]: why not mount the directory instead?

File: Makefile [hunk 3]
```diff
@@ -149,30 +149,27 @@ install-gometalinter:
...
+  endif
+
+.PHONY: build-goreleaser-image
+build-goreleaser-image: check-for-goreleaser-github-token
+	# Build custom goreleaser docker image using m3 directory root as context.
+	docker build -t m3/goreleaser -f ./docker/goreleaser/Dockerfile .
 
 .PHONY: release
-release: install-goreleaser
+release: build-goreleaser-image
```
Comment (prateek) [inline Makefile]: need to update the `PHONY` indication above this

Comment (richardartoul) [reply Makefile]: ...what does .PHONY do and why did this work?

File: docker/goreleaser/Dockerfile (hunk 2 above)
Comment (richardartoul) [reply docker/goreleaser/Dockerfile]: I think this was a hack to get around trying to mount a volume on root, I think I know how to fix it now though

File: Makefile (hunk 3 above)
Comment (richardartoul) [reply Makefile]: wait no this is fine, build-gorelease-iamge is just a dep

Comment (prateek) [reply Makefile]: d'oh. fair.

File: Makefile [hunk 4]
```diff
@@ -149,30 +149,27 @@ install-gometalinter:
...
+check-for-goreleaser-github-token:
+  ifndef GITHUB_TOKEN
+		echo "Usage: make GITHUB_TOKEN=\"<TOKEN>\" release"
+		exit 1
+  endif
+
+.PHONY: build-goreleaser-image
+build-goreleaser-image: check-for-goreleaser-github-token
+	# Build custom goreleaser docker image using m3 directory root as context.
+	docker build -t m3/goreleaser -f ./docker/goreleaser/Dockerfile .
```
Comment (prateek) [inline Makefile]: tag a version too

File: Makefile [hunk 5]
```diff
@@ -149,30 +149,27 @@ install-gometalinter:
...
 
 .PHONY: release
-release: install-goreleaser
+release: build-goreleaser-image
 	@echo Releasing new version
-	@source $(GO_BUILD_LDFLAGS_CMD) > /dev/null && goreleaser
+	docker run -e "GITHUB_TOKEN=$(GITHUB_TOKEN)" -v $(PWD):/m3 m3/goreleaser release
```
Comment (prateek) [inline Makefile]: pin to a version

File: Makefile [hunk 6]
```diff
@@ -149,30 +149,27 @@ install-gometalinter:
...
 
 .PHONY: release-snapshot
-release-snapshot: install-goreleaser
+release-snapshot: build-goreleaser-image
 	@echo Creating snapshot release
-	@source $(GO_BUILD_LDFLAGS_CMD) > /dev/null && goreleaser --snapshot --rm-dist
+	docker run -e "GITHUB_TOKEN=$(GITHUB_TOKEN)" -v $(PWD):/m3 m3/goreleaser --snapshot --rm-dist
```
Comment (prateek) [inline Makefile]: here too

**Pull request merged**